lib/super_digit.rb

```diff
@@ -1,15 +1,23 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n^2)
 def super_digit(n)
-   
+  return n.to_i if n.to_s.length == 1
+  digits = n.to_s.chars.map(&:to_i)
+  sum = 0
+  digits.each do |digit|
+    sum += digit
+  end
+  return sum if sum.to_s.length == 1
```

Comment:
Rather than converting it to a string, why not just 

```suggestion
  return sum if sum < 10
```